torch/_inductor/graph.py

```diff
@@ -251,15 +251,7 @@ def disable_cpp_wrapper(self, cond):
         assert not self.aot_mode, "AOT compilation failed"
         log.debug("Set cpp_wrapper to False due to %s", cond)
 
-    def check_buffer_for_cpp_wrapper(self, buffer: ir.ComputedBuffer):
```

Comment:
For `ExternKernel` which is not yet supported by the cpp wrapper, such as `ConvUnary` (WIP in https://github.com/pytorch/pytorch/pull/94964), this check here will disable the cpp wrapper if `ConvUnary` is present. If we remove the check here, it seems that cpp wrapper will still be enabled while it does not support this ExternKernel yet.

Reply:
Can we check if the `kernel` is `None` in the `register_buffer` instead?

Reply:
It seems that unsupported `ExternKernel` might have `kernel` which is not `None` as well:
https://github.com/pytorch/pytorch/blob/6145964ec9a556bc8bfd9a41024fed5501b9c5b6/torch/_inductor/ir.py#L3317-L3325

Reply:
In general, I don't like the way that disable_cpp_wrapper implicitly falls back to the default wrapper codegen. When it fails, it should fail loudly. Do you have an actual use case where you need this fallback behavior? If not, we should just assert in disable_cpp_wrapper.

Reply:
Here is one case that I have in mind: for example, for a model that has been splitted into several sub-graphs by dynamo, if one of the sub-graphs contains unsupported nodes, with the fallback, users could still use the cpp wrapper for sub-graphs that don't have such nodes. Otherwise, users won't be able to use the cpp wrapper at all until we make cpp wrapper fully mature.